--- hercules-2.16.2/configure.ac~	Mon May 20 23:05:09 2002
+++ hercules-2.16.2/configure.ac	Sun Jun 23 16:38:38 2002
@@ -279,31 +279,6 @@
                esac],
               [test x$debug != xtrue && auto_optimize=true])
 
-if test x$auto_optimize = xtrue; then
-  case "$host_cpu" in
-	  i586|i686)
-	    if test "$GCC" = "yes"; then
-		OFLAGS="-O3 -march=$host_cpu $OFLAGS"
-                if test x$enable_profiling != xtrue && test x$debug != xtrue; then
-                  OFLAGS="-fomit-frame-pointer $OFLAGS"
-                fi
-	    else
-                OFLAGS="-O2 $OFLAGS"
-	    fi
-	    ;;
-	  *)
-	    OFLAGS="-O2 $OFLAGS"
-	    ;;
-  esac
-fi
-
-if test "x$OFLAGS" = x; then
-  AC_MSG_RESULT([(none)])
-else
-  AC_MSG_RESULT([$OFLAGS])
-  CFLAGS="$OFLAGS $CFLAGS"
-fi
-
 AC_ARG_ENABLE(external-gui,
               AC_HELP_STRING([--disable-external-gui],
                              [disable interface to external Windows GUI]),
